src/test/java/com/marklogic/spark/writer/WriteArchiveOfFailedDocumentsTest.java

```diff
@@ -46,6 +53,27 @@ void test(@TempDir Path tempDir) {
         verifyArchiveRows(rows);
     }
 
+    /**
+     * Verifies that the success count and failure counts are properly incremented.
+     */
+    @Test
+    void singlePartition(@TempDir Path tempDir) {
+        MarkLogicWrite.setSuccessCountConsumer(count -> assertEquals(1, count));
+        MarkLogicWrite.setFailureCountConsumer(count -> assertEquals(3, count));
```

Comment:
Well, that's fascinating.
What if there's a bug in which one of those methods don't get called?

Reply:
Good point, I'll capture the count instead and do the assertion afterwards.